### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by jeward414**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -30,8 +36,20 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println(convertToJsonUsingGson(comments));
+    
+    Query query = new Query("Task");
```
Comment (sperzion) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:40]: Should this be called "Comment" instead?
Also you should probably set this up as a Java constant.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -30,8 +36,20 @@
...
+    DatastoreService dataStore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = dataStore.prepare(query);
+
+    ArrayList<String> commentList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        String comment = (String) entity.getProperty("comment-field");
```
Comment (sperzion) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Use a Java constant for the property name. That way you can change the name of it in one place.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -30,8 +36,20 @@
...
-    response.setContentType("text/html;");
-    response.getWriter().println(convertToJsonUsingGson(comments));
+    
+    final Query allComments = new Query("Comment");
+    DatastoreService dataStore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = dataStore.prepare(allComments);
+
+    ArrayList<String> commentList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        final String comment = (String) entity.getProperty("comment-field");
```
Comment (Google-JM) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Something to think about: What would happen in this code if `entity.getProperty("comment-field")` returns a null? 
Will this cause an error, or would the code function correctly?

Comment (Google-JM) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Create a global constant for any string you would use in getProperty(). This helps with reuse and preventing typo errors.

`public static final COMMENT_FIELD = "comment-field";`
`entity.getProperty(COMMENT_FIELD);`

Comment (jeward414) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: When I click submit when nothing is entered it behaves as usual. Right now I'm trying to add more fields, like a name field and a date field and a timestamp, what would be the best way to add all these fields?

Comment (sperzion) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Timestamp is probably something you want to generate on the server side. The simplest way is to use `System.getCurrentTimeMillis()`.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -22,40 +28,59 @@
...
+  private static final String COMMENT_FIELD = "comment-field";
+  private static final String NAME_FIELD = "name-field";
+  private static final String DATE_FIELD = "timestampValue";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println(convertToJsonUsingGson(comments));
+    
+    final Query allComments = new Query("Comment");
```
Comment (sperzion) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: Should also have a constant for "Comment"

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -156,6 +150,23 @@ <h3 class="centerheader">Photo Gallery</h3>
             <img src="/images/photog/three.JPG" style="width:100%">
         </div>
     </div>
-    
+    <div class="row">
+        <div class="photocolumn">
+        <form action="/data" method="POST">
+            <p> Enter your name along with a comment! </p>
+            <textarea name="name-field" placeholder="Enter Name"></textarea>
```
Comment (sperzion) [inline portfolio/src/main/webapp/index.html:157]: nit: lower case name, to be consistent with your comment prompt.

**Pull request merged**